lib/include/up-client-zenoh-cpp/rpc/zenohRpcClient.h

```diff
@@ -25,70 +25,122 @@
+
+            /**
+            * Terminates the zenoh RPC client  - the API should be called by any class that called init
+            * @return Returns OK on SUCCESS and ERROR on failure
+            */
+            uprotocol::v1::UStatus term() noexcept; 
+
+            /**
+            * API for clients to invoke a method (send an RPC request) and receive the response (the returned 
+            * {@link CompletionStage} {@link UPayload}. <br>
+            * Client will set method to be the URI of the method they want to invoke, 
+            * payload to the request message, and attributes with the various metadata for the 
+            * method invocation.
+            * @param topic The method URI to be invoked, ex (long form): /example.hello_world/1/rpc.SayHello.
+            * @param payload The request message to be sent to the server.
+            * @param options RPC method invocation call options, see {@link CallOptions}
+            * @return Returns the future 
+            */
+            std::future<uprotocol::rpc::RpcResponse> invokeMethod(const uprotocol::v1::UUri &topic, 
+                                                                  const uprotocol::utransport::UPayload &payload, 
+                                                                  const uprotocol::v1::CallOptions &options) noexcept;
+
+            /**
+            * API for clients to invoke a method (send an RPC request) and receive the response (the returned 
+            * {@link CompletionStage} {@link UPayload}. <br>
+            * Client will set method to be the URI of the method they want to invoke, 
+            * payload to the request message, and attributes with the various metadata for the 
+            * method invocation.
+            * @param methodUri The method URI to be invoked, ex (long form): /example.hello_world/1/rpc.SayHello.
+            * @param requestPayload The request message to be sent to the server.
+            * @param options RPC method invocation call options, see {@link CallOptions}
+            * @param callback that will be called once the future is complete
+            * @return UStatus
+            */
+            uprotocol::v1::UStatus invokeMethod(const uprotocol::v1::UUri &topic,
+                                                const uprotocol::utransport::UPayload &payload,
+                                                const uprotocol::v1::CallOptions &options,
+                                                const uprotocol::utransport::UListener &callback) noexcept;
+
+            /**
+             * get the number of max concurrent request 
+             * @return number of concurrent requests
+            */
+            size_t getMaxConcurrentRequests() {
+                return maxNumOfCuncurrentRequests_;
+            }
+            
+            /**
+             * get queue size
+             * @return queue size
+            */
+            size_t getQueueSize() {
```

Comment:
Should this go into the Zenoh spec to generally be shared among the implementations? Not the specific number of them, but the concept itself.

Reply:
@stevenhartley  what do you think?

Reply:
@stevenhartley ping

Reply:
@stevenhartley  the queue size means how many requests the uE can issue simultaneously, if the queue is full ,it will return an error

Reply:
@MelamudMichael if you expose this logic to the app developer, they need to maintain a list/queue of requests as well, I don't think we should do this.